src/reader-sync-browser.js

```diff
@@ -0,0 +1,158 @@
+import * as DecoderSync from './decoder-sync.js'
+
+/**
+ * @typedef {import('multiformats').CID} CID
+ * @typedef {import('./api').Block} Block
+ * @typedef {import('./api').CarReader} CarReaderIface
+ * @typedef {import('./coding').BytesReader} BytesReader
+ * @typedef {import('./coding').CarHeader} CarHeader
+ * @typedef {import('./coding').CarV2Header} CarV2Header
+ */
+
+/**
+ * Provides blockstore-like access to a CAR.
+ *
+ * Implements the `RootsReader` interface:
+ * {@link CarReader.getRoots `getRoots()`}. And the `BlockReader` interface:
+ * {@link CarReader.get `get()`}, {@link CarReader.has `has()`},
+ * {@link CarReader.blocks `blocks()`} (defined as a `BlockIterator`) and
+ * {@link CarReader.cids `cids()`} (defined as a `CIDIterator`).
+ *
+ * Load this class with either `import { CarReader } from '@ipld/car/reader'`
+ * (`const { CarReader } = require('@ipld/car/reader')`). Or
+ * `import { CarReader } from '@ipld/car'` (`const { CarReader } = require('@ipld/car')`).
+ * The former will likely result in smaller bundle sizes where this is
+ * important.
+ *
+ * @name CarReaderSync
+ * @class
+ * @implements {CarReaderIface}
+ * @property {number} version The version number of the CAR referenced by this
+ * reader (should be `1` or `2`).
+ */
+export class CarReaderSync {
+  /**
+   * @constructs CarReader
+   * @param {CarHeader|CarV2Header} header
+   * @param {Block[]} blocks
+   */
+  constructor (header, blocks) {
+    this._header = header
+    this._blocks = blocks
+    this._keys = blocks.map((b) => b.cid.toString())
+  }
```

Comment:
Can we have a method to perform verification that will verify that blocks match claimed CIDs ?

Reply:
@Gozala this would be worthwhile but needs to be an option and should be tackled in a separate PR. Opened an issue here and it'd be great if someone could pick this off: https://github.com/ipld/js-car/issues/123

As noted there, this requires having the hashers available to do the validation, which is always a problem in JS—what do you do if someone shows up with blake2b CIDs?

Also worth having notes in the README, although pretty much all of the JS stack doesn't perform this kind of validation, so there's a bit of a caveat emptor all round on that front unfortunately.